src/eliminate_contiguous.cpp

```diff
@@ -51,6 +51,14 @@ static bool try_compute_shape(instruction_ref ins,
             return false;
         }
 
+        // If the instruction is a gemm and any of the inputs is broadcasted, the input needs to be
+        // made contiguous
+        if(contains(ins->name(), "gemm") and
+           any_of(inputs, [](const auto& sh) { return sh.broadcasted(); }))
+        {
+            return false;
+        }
```

Comment:
This logic should be handled in the `compute_shape` method of the gemm operator. It should throw an error for this case if it can't handle such shapes.

Reply:
Done.